@here/olp-sdk-core/lib/utils/MultiPartUpload.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import {
+    DataStoreRequestBuilder,
+    HRN,
+    OlpClientSettings,
+    RequestFactory
+} from "@here/olp-sdk-core";
+import { BlobApi, ObjectStoreApi } from "@here/olp-sdk-dataservice-api";
+
+interface MultiPartUploadParams {
+    /**
+     * The instance of OlpClientSettings.
+     */
+    settings: OlpClientSettings;
+    /**
+     * The version od Blob API to upload.
+     */
+    blobVersion: "v1" | "v2";
+    /**
+     * The Layer ID to upload.
+     */
+    layerId: string;
+    /**
+     * If you're uploading to Blob V1, set the datahandle
+     * and set the key for Blob V2.
+     */
+    keyOrDatahandle: string;
```

Comment:
Maybe we should name this more common, e.g. `handle` and leave it up to the comment to make sure the user provides the correct type.